deb-package/build

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+#
+# Builds a Debian package for MXNET
+#
+# Copyright sociomantic labs GmbH 2017.
+# Distributed under the Apache License, Version 2.0.
+# (See accompanying LICENSE file or copy at
+# https://www.apache.org/licenses/LICENSE-2.0.txt)
+set -xeu
+
+# Set up package
+pkgname=libmxnet
+pkgversion_number=$(git describe --tags --dirty |
+                    sed 's/^v\(.*\)$/\1/' |
+                    sed 's/-\([0-9]\+\)-\(g[0-9a-f]\+\)/+\1~\2/' |
+                    sed 's/\(~g[0-9a-f]\+\)-dirty$/-dirty\1/' |
+                    sed 's/-dirty/~dirty.'`date +%Y%m%d%H%M%S`'/'
+                   )
+pkgdistro=$(lsb_release -cs)
+pkgversion="${pkgversion_number}~${pkgdistro}"
```

Comment:
I think there is quite some confusion here on how versions are handled by SemVer and Debian. In SemVer `-` is to mark a "pre-release", in Debian is a separator to differentiate between the upstream version and the debian build (for example, if upstream version v1.2.3 is packaged for the first time, then the debian version will probably be something like 1.2.3-0. Suppose that the pre-install script in the package had a bug, and you want to build another package, then you fix the script in the package and make the version 1.2.3-1. The upstream version is the exact same, is just the "package" version that changes. There is no equivalence for this in SemVer.
Then in SemVer `+` is used to attach build-metadata, which is sort of similar to the previous but not the same, because in terms of version calculation (if a version is greater than another) is a NOP. In Debian the build-metadata is specified by `~`.

Seeing this, it might make more sense to use the `~` to specify the distro in the package version, that is a good point, but having `pkgversion="${pkgversion_number}~${pkgdistro}"` doesn't make a lot of sense IMHO, because this now isn't the changing the package version part but the upstream version part, if there is no `-` in the Debian version (and relying in tagging with `+socio1` might not be reliable).

Also the conversion between `+socioN` to `-socioN` changes the meaning in the version, as the former is metadata that doesn't participate in version calculation while the later isn't.

In any case, what it clear is that this is tricky. So I'm more inclined than before that we should use Makd to make these conversions.